### PR TITLE
Update bazel files to build and run examples

### PR DESCRIPTION
**Pull request by RMeli**
*Disclaimer*: this is my very first time using [bazel](https://bazel.build/). I do not (entirely) know what I'm doing.

This PR partially addresses #250 with the following:

* Update headers in `autodiff/BUILD`
* Update executables in `examples/*/BUILD`
* Remove stale `test/BUILD` directory
* Add [bazel](https://bazel.build/) to CI [only builds and runs examples]
* Add scarce instructions on how to use [bazel](https://bazel.build/) in [autodiff.github.io/installation/](https://autodiff.github.io/installation/) to build and run the examples

This PR **does not** allow to build and run the unit tests with [bazel](https://bazel.build/). This will be addressed in a later PR, together with proper installation instructions using [bazel](https://bazel.build/) as build system.

---

Edit: I was meant to open this PR against my own fork, oh well...

### CODE REVIEW

**Comment by allanleal** [comment]
Hi @RMeli , thanks for this.

> Edit: I was meant to open this PR against my own fork, oh well...

The bazel builds are passing. All looks ok to me. Is there anything else you want to do before I merge?

File: docs/website/installation.md [hunk 1]
```diff
@@ -68,6 +68,34 @@ systems. To install {{autodiff}} locally, use:
...
+~~~
+bazel build //examples/forward:all
+bazel build //examples/reverse:all
+~~~
+
+Run all examples using [bazel](https://bazel.build/) and display their output:
+
+~~~
+bazel test //examples/forward:all
+bazel test //examples/reverse:all
```
Comment (RMeli) [inline docs/website/installation.md:96]:
```suggestion
bazel test //examples/forward:all --test_output=all
bazel test //examples/reverse:all --test_output=all
```

File: examples/forward/BUILD [hunk 2]
```diff
@@ -1,100 +1,118 @@
 cc_test(
-    name = "example-forward-single-variable-function",
-    srcs=["example-forward-single-variable-function.cpp"],
+    name = "example-forward-directional-taylor-series-using-eigen",
+    srcs=["example-forward-directional-taylor-series-using-eigen.cpp"],
```
Comment (RMeli) [inline examples/forward/BUILD:3]:
```suggestion
    srcs = ["example-forward-directional-taylor-series-using-eigen.cpp"],
```

File: examples/forward/BUILD [hunk 3]
```diff
@@ -1,100 +1,118 @@
...
     deps = ["//:autodiff"],
     copts = ["-I./"]
 )
 
 cc_test(
-    name = "example-forward-single-variable-function-custom-scalar",
-    srcs = ["example-forward-single-variable-function-custom-scalar.cpp"],
+    name = "example-forward-directional-taylor-series",
+    srcs=["example-forward-directional-taylor-series.cpp"],
```
Comment (RMeli) [inline examples/forward/BUILD:10]:
```suggestion
    srcs = ["example-forward-directional-taylor-series.cpp"],
```

File: examples/forward/BUILD [hunk 4]
```diff
@@ -1,100 +1,118 @@
...
-    srcs = ["example-forward-gradient-derivatives-using-eigen-with-parameters.cpp"],
+    name = "example-forward-multi-variable-function-with-parameters",
+    srcs = ["example-forward-multi-variable-function-with-parameters.cpp"],
     deps = ["//:autodiff"],
     copts = ["-I./"]
 )
 
+cc_test(
+    name = "example-forward-multi-variable-function",
+    srcs=["example-forward-multi-variable-function.cpp"],
```
Comment (RMeli) [inline examples/forward/BUILD:101]:
```suggestion
    srcs = ["example-forward-multi-variable-function.cpp"],
```

File: examples/forward/BUILD [hunk 5]
```diff
@@ -1,100 +1,118 @@
...
+cc_test(
+    name = "example-forward-single-variable-function-custom-scalar",
+    srcs = ["example-forward-single-variable-function-custom-scalar.cpp"],
+    deps = ["//:autodiff"],
+    copts = ["-I./"]
+)
 
+cc_test(
+    name = "example-forward-single-variable-function",
+    srcs=["example-forward-single-variable-function.cpp"],
```
Comment (RMeli) [inline examples/forward/BUILD:115]:
```suggestion
    srcs = ["example-forward-single-variable-function.cpp"],
```

File: examples/reverse/BUILD [hunk 6]
```diff
@@ -1,3 +1,10 @@
+cc_test(
+    name = "example-reverse-conditional",
+    srcs=["example-reverse-conditional.cpp"],
```
Comment (RMeli) [inline examples/reverse/BUILD:3]:
```suggestion
    srcs = ["example-reverse-conditional.cpp"],
```

**Comment by RMeli** [comment]
Everything should be good to go once CI comes back green again.

**Comment by allanleal** [comment]
Thanks a lot, Rocco!

**Pull request merged**